### DISCUSSION TITLE
problema con il servizio climate.turn_on

### DISCUSSION THREAD

**Discussion by alfry82** [indomus/forum#1802] [HUB: Home Assistant]
Buongiorno a tutti,

sto riscontrando un problema con il servizio climate.turn_on in quanto quando chiamo il servizio il condizionatore si accende sempre in modalità riscaldamento.
ho configurato 3 condizionatori su HA ( uno con l'integrazione SmartIR ed altri due con un componente personalizzato) da circa 2 anni.

Utilizzo alexa per comandare i condizionatori ("Alexa accendi condizionatore...)

Quando passavo da inverno ad estate o viceversa, per far si che alexa accendesse il condizionatore nella modalità corretta, utilizzavo l'interfaccia utente di HA accendendo il condizionatore nella modalità desiderata e rispegnendolo ( a volte dovevo farlo piu di una volta affiche funzionasse correttamente)
da questa estate sono riuscito a far accendere automaticamente nella modalità COOL solo il condizionatore con integrazione SMARTIR e non gli altri due.

provando a chiamare il servizio dalla maschera "strumenti per sviluppatori" ho notato che il servizio climate.turn_on imposta in automatico su CALDO.
C'è un modo per impostare una modalità di default?

**Comment by snakuzzo** [comment]
https://www.home-assistant.io/integrations/climate/#service-climateset_hvac_mode

**Comment by alfry82** [comment]
avevo già provato questo servizio ma chiamandolo accende effettivamente il condizionatore nella modalità corretta ma non risolve il problema dell'avvio successivo.
il condizionatore si riaccenderà in modalità Riscaldamento utilizzando alexa o il servizio climate.turn_on

**Reply by snakuzzo** [reply]
Io ho appena fatto una prova con il climate creato automaticamente dall'integrazione Daikin.
Chiamando solo il servizio `climate.turn_on` il clima si accende impostato nella modalità in cui era prima dello spegnimento.

**Reply by alfry82** [reply]
> Io ho appena fatto una prova con il climate creato automaticamente dall'integrazione Daikin. Chiamando solo il servizio `climate.turn_on` il clima si accende impostato nella modalità in cui era prima dello spegnimento.

Si questo è il comportamento atteso, ma non capisco perchè a me non funziona cosi, qualche consiglio su come potrei risolvere il problema?

**Reply by snakuzzo** [reply]
premesso che andrebbe visto nel dettalgio...ma se per qualche strana causa irrisolvibile non funziona, dovresti salvarti ogni volta lo stato (hvac) attuale da qualche parte (anche un input text va bene) e poi ad ogni accensione fai scattare un'automazione che imposta l'hvac mode con la modalità salvata nell'input text. Così facendo ogni volta che accendi il clima, si accenderà nella modalità in cui era prima delo spegnimento

**Reply by alfry82** [reply]
si avevo già scritto un automazione ma cosi è un po sporca perchè il condionatore si accende in modalità riscaldamento e dopo qualche secondo viene impostato a raffreddamento. Cosi facendo noto che il condizionatore rimane fermo per un po (anche 5 minuti e poi parte)
non vorrei rischiare di danneggiarlo.
Potrei provare a fare un istallazione pulita ma sarebbe un lavoraccio ripristinare tutto e non so se alla fine risolverei il problema.

Grazie comunque per la disponibilità

**Reply by snakuzzo** [reply]
l'installazione pulita non serve...è già pulita di suo.
In tutto ciò io non ho capito quello che ti dà il comportamento "indesiderato" come l'hai integrato su HA

**Reply by alfry82** [reply]
Come integrazione ho utilizzato il componente personalizzato 
https://community.home-assistant.io/t/add-support-for-fujitsu-wireless-air-conditioning-control-app-fglair/65999
che per un paio di stagioni non mi ha dato problemi, poi è all'inizio di questa estate che avrei voluto usare l'accensione a raffreddamento mi sta dando questo problema, si accende di default a riscaldamento.

Pensando che fosse un problema legato al componente personalizzato ho provato ad utilizzare un'altra integrazione che utilizza MQTT:
https://github.com/deiger/AirCon#prerequisites
ma riscontro lo stesso problema 

Ho anche un altro condizionatore che piloto con un RM MINI 3 e con quello non ho problemi, è bastato accenderlo una volta a Raffreddamento e da allora climate.turn_on lo accende direttamente a freddo.

**Reply by snakuzzo** [reply]
Allora non credo proprio sia un problema di HA...
Dovresti aprire le issue su quei due progetti.
